src/library/librarycontrol.h

```diff
@@ -65,27 +79,51 @@ class LibraryControl : public QObject {
     void slotDecrementFontSize(double v);
 
   private:
-    Library* m_pLibrary;
-
-    ControlObject* m_pSelectNextTrack;
-    ControlObject* m_pSelectPrevTrack;
-    ControlObject* m_pSelectTrack;
-
-    ControlObject* m_pSelectSidebarItem;
-    ControlObject* m_pSelectPrevSidebarItem;
-    ControlObject* m_pSelectNextSidebarItem;
-
-    ControlObject* m_pToggleSidebarItem;
-    ControlObject* m_pLoadSelectedIntoFirstStopped;
-    ControlObject* m_pAutoDjAddTop;
-    ControlObject* m_pAutoDjAddBottom;
-
-    ControlObject* m_pFontSizeKnob;
-    ControlPushButton* m_pFontSizeIncrement;
-    ControlPushButton* m_pFontSizeDecrement;
-
-    WLibrary* m_pLibraryWidget;
-    WLibrarySidebar* m_pSidebarWidget;
+    Library* m_pLibrary = nullptr;
+
+    // Simulate pressing a key on the keyboard
+    void emitKeyEvent(QKeyEvent&& event);
+
+    // Controls to navigate vertically within currently focussed widget (up/down buttons)
+    ControlPushButton* m_pMoveUp = nullptr;
```

Comment:
> In this case to follow that rule I guess we should try to move all of the controller variable initializations to the initialization list.

Yes, we could do it, but I am not a big fan of it, because it produces some clutter because, the instantiation and the setup are spitted. In most places in the existing Mixxx code the new() call and the connections setup are side by side. 

There is an other thing, that bugs me more: How to atomize the live time handling of the ControlObjects. The most effective way is to use the QT object tree. This is already implemented for the ControlProxy, and should be added here as well. But that is a different story.